### ISSUE TITLE
[docs] Switch code samples between TypeScript & JavaScript

### ISSUE THREAD

**Issue by ghost** [lit/lit-element#550]
User should be able to choose whether to see TypeScript or JavaScript code samples

**Comment by fila90** [comment]
hi,

not sure if this is the right place to put this, and correct me if I'm wrong but I think you switched examples [here](https://lit-element.polymer-project.org/)
TS code editor opens up `.ts` file that's written in vanilla `js`
JS code editor opens up `.js` file that's written in `TypeScript`

also if you could disable `"editor.scrollBeyondLastLine"` in VS Code, that would be great.

Cheers.

**Comment by katejeffreys** [comment]
Thanks for the heads-up, 

* the TS example might not use TypeScript correctly because I made it and I don't know how to TypeScript properly yet. I should fix that, & the code :) Maybe @justinfagnani or @arthurevans could help me out with some correct TypeScript for this sample.

* When I click "Launch code editor (JavaScript)" it shows the code below, which looks like vanilla JS to me..? What code are you seeing after clicking "Launch code editor (JavaScript)"?

```js
import { LitElement, html } from 'lit-element';

class SimpleGreeting extends LitElement {
  static get properties() {
    return { name: { type: String } };
  }

  constructor() {
    super();
    this.name = 'World';
  }
  
  render() {
    return html`<p>Hello, ${this.name}!</p>`;
  }
}

customElements.define('simple-greeting', SimpleGreeting);
```

* >if you could disable `"editor.scrollBeyondLastLine"` in VS Code, that would be great.

That would be good to do if possible. The embedded code samples use StackBlitz JavaScript sdk, and I'm not sure if there's a setting for that. I will try to find out.

**Comment by fila90** [comment]
Hi, 
thanks for getting back.

I'm also not proficient in Typescript, and I might be completely wrong here, but looking at the docs of Typescript they have bunch of types declared all around. In your code I just see you use these 2 decorators `@property()` and `@customElement()`

Vanilla JS looks the same as you described, but can you please explain what does 
``` javascript
  static get properties() {
    return { name: { type: String } };
  }
```
do? Removing that line doesn't seem to affect the app.
Also, on last line you use `customElements` without importing it like in `.ts` example. How does this work?

**Comment by arthurevans** [comment]
I think the editors are coming up with the correct versions. It can be a little confusing if you click both of them one after the other, though, since it takes a moment for the editor to render.

In our JavaScript samples, we try to stick to the widely-supported version of JavaScript... That is, something that can run without transformation on modern browsers—right now, that's ES2015, which is supported in most evergreen browsers.

TypeScript has decorators built-in. I believe an EcmaScript decorators proposal has now reached some level of consensus, but it's not widely-supported in browsers yet, so you need to use something like Babel to use decorators in JavaScript. So that's why we show decorators in the TS sample and not JS.

**Comment by katejeffreys** [comment]
@fila90 you might have already but for future readers I'll put the rest of the answers anyway

* `static get properties` - see [Declare properties](https://lit-element.polymer-project.org/guide/properties#declare) - is where you list the properties that will cause an update to DOM when changed. In that simple example, failing to declare the type doesn't break anything but the `type` can be important when [converting between attributes (always strings) and non-string properties](https://lit-element.polymer-project.org/guide/properties#conversion). 

As you mentioned, `@property({...})` is a shortcut & does the same thing as `static get properties()...` 

* The import for `customElement` in the typescript example is importing the `@customElement(...)` decorator, not the [`customElements` registry](https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/define), which is part of the web platform & implemented by all browsers except IE & Edge. Most browsers don't need to import it, but for IE and Edge, we can do so by [loading the Web Components polyfills](https://lit-element.polymer-project.org/guide/use#polyfills).

* >not sure why are decorators bound to `TypeScript`, they should be just a shorthand to define property and element?

they will be when decorators are widely implemented i think... until then for the reasons Arthur said, we tend to show them in ts but not js examples. 

* >TypeScript example should be something like this `@property({type : String}) name = 'World';`

I think we had it like that in the draft, can't remember why we changed it. Maybe because `String` is the default type anyway so we were able to cut it to make the example simpler.

Anyway we will eventually be able to provide more TypeScript examples. In the mean time thanks for your comments, it's good to re-think assumptions about what is clear to readers.
